lib/Builder.php

```diff
@@ -198,4 +196,30 @@ private function render(string $directory, string $targetDirectory) : void
             new PostBuildRenderEvent($this, $directory, $targetDirectory)
         );
     }
+
+    private function loadCachedMetas(string $targetDirectory) : void
+    {
+        $metaCachePath = $this->getMetaCachePath($targetDirectory);
+        if (! file_exists($metaCachePath)) {
+            return;
+        }
+
+        $contents = file_get_contents($metaCachePath);
+
+        if ($contents === false) {
+            return;
```

Comment:
I think you want to throw an exception here. If `file_exists` returns true and `file_get_contents` returns false, something is wrong.

Reply:
Yes, an exception is what you want here and it will satisfy phpstan too.

Reply:
It is technically possible for a systems permissions to return true for file exists but then false when you try to read so you want to let the end user know that with an exception instead of silently ignoring it.